src/teacher.py

```diff
@@ -38,36 +46,15 @@ def _answer_GroundAtomsHold_query(
             holds[ground_atom] = pred.holds(state, ground_atom.objects)
         return GroundAtomsHoldResponse(query, holds)
 
-
-@dataclass(frozen=True, eq=False, repr=False)
-class Query(abc.ABC):
-    """Base class for a Query.
-
-    Has no API.
-    """
-
-
-@dataclass(frozen=True, eq=False, repr=False)
-class Response(abc.ABC):
-    """Base class for a Response to a query.
-
-    All responses contain the Query object itself, for convenience.
-    """
-    query: Query
-
-
-################################################################################
-#                        Query and Response subclasses                         #
-################################################################################
-
-
-@dataclass(frozen=True, eq=False, repr=False)
-class GroundAtomsHoldQuery(Query):
-    """A query for whether ground atoms hold in the state."""
-    ground_atoms: Collection[GroundAtom]
-
-
-@dataclass(frozen=True, eq=False, repr=False)
-class GroundAtomsHoldResponse(Response):
-    """A response to a GroundAtomsHoldQuery, providing boolean answers."""
-    holds: Dict[GroundAtom, bool]
+    def _answer_Demonstration_query(
+            self, state: State,
+            query: DemonstrationQuery) -> DemonstrationResponse:
+        task = Task(state, query.goal)
+        try:
+            policy = self._oracle_approach.solve(task, CFG.timeout)
+        except (ApproachTimeout, ApproachFailure):
+            return DemonstrationResponse(query, teacher_traj=None)
+
+        teacher_traj, _, _ = utils.run_policy_on_task(
```

Comment:
do we want to assign the third return value to a variable `goal_reached` and then `assert goal_reached` in the next line? i can't imagine the teacher should ever formulate a plan that doesn't actually work